test/upgrade/prober/configuration.go

```diff
@@ -20,63 +20,74 @@ import (
 	"context"
 	"fmt"
 	"io/ioutil"
+	"os"
 	"path"
 	"runtime"
+	"strings"
 	"text/template"
 	"time"
 
 	"github.com/kelseyhightower/envconfig"
 	"github.com/wavesoftware/go-ensure"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
-	testlib "knative.dev/eventing/test/lib"
-	"knative.dev/eventing/test/lib/duck"
 	"knative.dev/eventing/test/lib/resources"
-	"knative.dev/pkg/apis"
+	"knative.dev/eventing/test/upgrade/prober/sut"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	pkgupgrade "knative.dev/pkg/test/upgrade"
 )
 
 const (
 	defaultConfigName          = "wathola-config"
 	defaultConfigHomedirPath   = ".config/wathola"
 	defaultHomedir             = "/home/nonroot"
 	defaultConfigFilename      = "config.toml"
-	defaultWatholaEventsPrefix = "com.github.cardil.wathola"
-	defaultBrokerName          = "default"
+	defaultWatholaEventsPrefix = "dev.knative.eventing.wathola"
 	defaultHealthEndpoint      = "/healthz"
 	defaultFinishedSleep       = 5 * time.Second
 
 	Silence DuplicateAction = "silence"
 	Warn    DuplicateAction = "warn"
 	Error   DuplicateAction = "error"
+
+	prefix = "eventing_upgrade_tests"
+
+	// TODO(ksuszyns): Remove this val in next release
+	// Deprecated: use 'eventing_upgrade_tests' prefix instead
+	deprecatedPrefix = "e2e_upgrade_tests"
 )
 
 // DuplicateAction is the action to take in case of duplicated events
 type DuplicateAction string
 
-var eventTypes = []string{"step", "finished"}
-
 // Config represents a configuration for prober.
 type Config struct {
 	Wathola
-	Namespace     string
 	Interval      time.Duration
 	FinishedSleep time.Duration
 	Serving       ServingConfig
 	FailOnErrors  bool
 	OnDuplicate   DuplicateAction
-	BrokerOpts    []resources.BrokerOption
+	Ctx           context.Context
+	// BrokerOpts holds opts for broker.
+	// TODO(ksuszyns): Remove this opt in next release
+	// Deprecated: use Wathola.SystemUnderTest instead.
+	BrokerOpts []resources.BrokerOption
+	// Namespace holds namespace in which test is about to be executed.
+	// TODO(ksuszyns): Remove this opt in next release
+	// Deprecated: namespace is about to be taken from testlib.Client created by
+	// NewRunner
+	Namespace string
 }
 
 // Wathola represents options related strictly to wathola testing tool.
 type Wathola struct {
-	ConfigMap
+	ConfigToml
+	SystemUnderTest  sut.SystemUnderTest
```

Comment:
I would like to stay with SUT, as that a standard naming in testing world. We are here in Eventing space, serving will not use this, nor there will be no confusion.

Even, if Serving will follow this change, and prepare its own `SystemUnderTests` interface it would still be clear what is what. We have here full package of `knative.dev/eventing/test/upgrade/prober/sut`. Serving, if introduced, will be different for ex.: `knative.dev/serving/test/upgrade/prober/sut`. Even on integration places that would not be a problem as we just do aliasing as we regulary do:

```golang
// maybe in serverless-operator or knative-operator
import (
  eventingsut "knative.dev/eventing/test/upgrade/prober/sut"
  servingsut "knative.dev/serving/test/upgrade/prober/sut"
)
```